libraries/AP_HAL_Linux/RPIOUARTDriver.cpp

```diff
@@ -201,19 +173,7 @@ void RPIOUARTDriver::_timer_tick(void)
         n = _max_size;
     }
 
-    if (n > 0) {
-        uint16_t n1 = _writebuf_size - _writebuf_head;
-        if (n1 >= n) {
-            // do as a single write
-            memcpy( &((uint8_t *)_dma_packet_tx.regs)[0], &_writebuf[_writebuf_head], n );
-        } else {
-            // split into two writes
-            memcpy( &((uint8_t *)_dma_packet_tx.regs)[0], &_writebuf[_writebuf_head], n1 );
-            memcpy( &((uint8_t *)_dma_packet_tx.regs)[n1], &_writebuf[0], n-n1 );
-        }
-
-        BUF_ADVANCEHEAD(_writebuf, n);
-    }
+    _writebuf.read(&((uint8_t *)_dma_packet_tx.regs)[0], n);
```

Comment:
_dma_packet_tx.regs is uninitialized in some cases... valgrind will complain about uninitialized members being given to ioctl() when you call the _dev->transfer() method. you should either initialize _dma_packet_tx entirely or write _dma_packet_tx.regs to 0. Can we fix this while we are at it?